documentation/contributing/pullrequests.markdown

```diff
@@ -0,0 +1,85 @@
+The only commits that might be pushed directly are cherry-picks from older branches (see below) and administrative changes (e.g. CHANGELOGS, tags).
+
+### 1+1 == merge
+
+If you see a trustworthy "+1" note in a request or the requestor is a fellow maintainer and all feedback has been addressed by the requestor,
+merge the request after your own review. Otherwise add a "+1" note after you are satisfied after the review.
+
+### squash-merge if possible
+
+Pull-requests with a single commit should *always* be squash-merged (github supports this via a drop-down list on the "merge" button).
+If a request contains "fixup commits" it should also be squash-merged. If you are unsure this is to the requestor's liking, ask him.
+
+### feature branches in upstream repositories
+
+In general, requests should originate from a private fork of the upstream repository. If maintainers expect support in the development of a patch, they are free to create a new branch in the upstream repository instead. This enables other maintainers to directly push changes there and enables users to add pull-requests targeting the feature branch. To keep the list of branches clear and unambiguous, **names for such branches should always follow the pattern ``request-<keyword description>``**. This makes it clear these branches are not relevant to users not involved in the respective request.
+
+### the latest branch preserves all original commits
+
+All commits that get merged into MoveIt! have to end up in the latest branch "as is" in order to preserve the history of the changes.
+Future versions of MoveIt might not share the whole history of older branches and finding your `git blame` returns a commit called ``cherry-picked #123``, without an additional description, is not desirable.
+
+### merge only if you make sure it becomes available in later branches too
+
+A request must not be merged unless the merging maintainer makes sure the same (or a similar) patch-set is merged into all later branches at the same time.
+We don't want to loose changes in one ROS distribution because someone forgot to do that.
+If the commits apply cleanly to future branches and there is no evidence that they will break anything there, the maintainer merging the request is free to add the changes to the respective branches. (This has to respect the all-commits-in-latest-branch guideline though).
+Otherwise either the requestor or the maintainer should create new pull-requests targeting the later branches.
+If possible, these should be merged *together* with the original request.
+In this context it might be worth spending some time on using features available in later ROS distributions to simplify code in the request, e.g. by using a new coding standard or a more current version of a library.
+Experience shows that code simplification seldom happens after a working feature was merged once.
```

Comment:
this section is a bit complicated, maybe im tired of reading this doc by now. perhaps turn it into easier to read bullet points